### ISSUE TITLE
volto-slate does not properly support "list in list" as generated by Plone Classic

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#4538]
In Plone classic, create a list in a list in a richtext field. This is how do to it:

- create a list with a single line
- create another line in the list (another list item)
- increase the level of the list item

The generated HTML is:

```
<ul>
  <li>blabla
     <ul>
        <li>dasdasdas</li>
     </ul>
   </li>
</ul>
```
Notice that the sublist is a part of the list item.

If you try the same thing with Volto (you can increate the level of the elements using the Tab key at the beginning of the list item), you'll get the following markup:

```
<ul>
  <li>blabla</li>
  <ul>
        <li>dasdasdas</li>
   </ul>
</ul>
```

If you copy the rendered markup of the above list and paste it in a Volto block, it will "silently eat" the sublist. If you generate the proper slate structure (using code from the plone.volto slate support PR), everything will appear to be working, but when you try to do any editing on the text block, the sublist will be swallowed.

**Comment by tiberiuichim** [comment]
Here's a page with sublists as a sample: https://climate-adapt.eea.europa.eu/en/eu-adaptation-policy/strategy

**Comment by tiberiuichim** [comment]
this is "expected behavior" because `config.slate.inlineElements` does not contain either `ul` or `li` elements.

According to the TinyMCE generated markup, they should be inline.

**Comment by davisagli** [comment]
@tiberiuichim What do you mean that they should be inline according to the TinyMCE markup? The TinyMCE markup for nested lists is correct (there is a similar example at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) but lists and list items are block-level elements in HTML, not inline -- https://developer.mozilla.org/en-US/docs/Web/HTML/Block-level_elements

**Comment by tiberiuichim** [comment]
@davisagli I agree, it's just that I need to understand how to align the expectations of slate vs our internal model.

The problem we're hitting one of these two conditions: https://github.com/ianstormtaylor/slate/blob/1d8010be8500ef5c98dbd4179354db1986d5c8f4/packages/slate/src/create-editor.ts#L236-L237C13

Basically, for an element `<li>text<ul>...</ul></li>`, the first child (`node.children[0]`) is an inline element (Text), so slate expects that all the other children are also inline elements.

So, in my screenshot, when Slate is normalizing the parent `<li>` and it reaches the `<ul>` child, `isInlineOrText` is False for the `ul`, so it just removes it.

**Comment by tiberiuichim** [comment]
This comment from slate code is a bit head-scratching: 

https://github.com/ianstormtaylor/slate/blob/1d8010be8500ef5c98dbd4179354db1986d5c8f4/packages/slate/src/create-editor.ts#L253-L256

```
        // Only allow block nodes in the top-level children and parent blocks
        // that only contain block nodes. Similarly, only allow inline nodes in
        // other inline nodes, or parent blocks that only contain inlines and
        // text.
```

**Comment by davisagli** [comment]
@tiberiuichim I think it means we have to turn
```
<li>
  some text
  <ul> ... </ul>
</li>
```
into
```
<li>
  <div>some text</div>
  <ul> ... </ul>
</li>
```
so that the text is not parented by a node that contains both block-level and inline elements.

**Comment by tiberiuichim** [comment]
@davisagli as an additional piece of information, volto-slate uses almost exclusively Element nodes (blocks), but declares them to be "inline", for example https://github.com/plone/volto/blob/master/packages/volto-slate/src/editor/plugins/Link/extensions.js#L11
